### PR TITLE
Support WebSphere Liberty Operator

### LINKED ISSUE #65: Add support for WebSphere Liberty Operator
- Add a new tab, `Configure operator`, after `Configure application`
- Add param `IBM Supported`: yes/no (default: `no`)
- If `IBM Supported` is yes, display these three **required** params
    - Accept license: checkbox (default:  unchecked)
        - I accept the terms on the license agreement corresponding to the version of IBM Program
in the application container by setting this value to true. See https://ibm.biz/was-license 
for the license agreement applicable to this IBM Program. You must accept the license 
for the installation to work.

    - Edition: drop down with following 3 values (default: none selected)
        - "IBM WebSphere Application Server"
        - "IBM WebSphere Application Server Liberty Core"
        - "IBM WebSphere Application Server Network Deployment"
        - Help text: `Product edition`
    - Product entitlement source: drop down with following 4 values (default: none selected)
        - "Standalone"
        - "IBM WebSphere Hybrid Edition"
        - "IBM Cloud Pak for Applications"
        - "IBM WebSphere Server Family Edition"
       - Help text: `Entitlement source for the product`

Notes: 
- If `IBM Supported` is no, install Open Liberty Operator
- If `IBM Supported` is yes, install WebSphere Liberty Operator
- Use `Edition` and `Product entitlement source` values in app deploy yaml
- App deploy yaml also needs `Metric` parameter. Set it to "Processor Value Unit (PVU)” for "Standalone” and "IBM WebSphere Server Family Edition”, otherwise set it to "Virtual Processor Core (VPC)”.

### PR DESCRIPTION
**Pull request by majguo**
## Description

The PR is to simplify the user experience by resolving the following issues:
* Resolve #65

## Change summary

* Update UI to support install WebSphere Liberty Operator
* Update ARM templates to support WLO
* Update scripts to support WLO
* Add two more templates for support `WebSphereLibertyApplication` CR
* Update `integration-test` pipeline to support WLO

## Testing

Verified the following 8 test cases with `integration-test` pipeline (**Note**: For all cases enabled with **WLO**, **Edition** is `IBM WebSphere Application Server` and **Product entitlement source** is `Standalone`):
|AGIC |WLO|OLO|Sample App|Pipeline run|
|----- |------|-----|-------------|-------------|
|        |         |X     |                   |[integration-test-65](https://github.com/majguo/azure.liberty.aks/actions/runs/3341957166)|
|        |         |X     |X                 |[integration-test-66](https://github.com/majguo/azure.liberty.aks/actions/runs/3341960370)|
|X      |         |X     |                   |[integration-test-67](https://github.com/majguo/azure.liberty.aks/actions/runs/3341961520)|
|X      |         |X     |X                 |[integration-test-68](https://github.com/majguo/azure.liberty.aks/actions/runs/3341962468)|
|        |X       |       |                   |[integration-test-69](https://github.com/majguo/azure.liberty.aks/actions/runs/3342207085)|
|        |X       |       |X                 |[integration-test-70](https://github.com/majguo/azure.liberty.aks/actions/runs/3342209423)|
|X      |X       |       |                   |[integration-test-71](https://github.com/majguo/azure.liberty.aks/actions/runs/3342211302)|
|X      |X       |       |X                 |[integration-test-72](https://github.com/majguo/azure.liberty.aks/actions/runs/3342212611)|

Verified the following 5 test cases using [preview link](https://portal.azure.com/?feature.customportal=false#create/microsoft_javaeeonazure_test.20191212-arm-rhel-was-nd-v9-cluster-previewazure-liberty-aks):
|AGIC |WLO|OLO|Sample App|Edition|Product entitlement source|
|----- |------|-----|-------------|--------|------------------------------|
|X      |         |X     |X                 |IBM WebSphere Application Server|Standalone|
|X      |X       |       |X                 |IBM WebSphere Application Server|Standalone|
|        |X       |       |X                 |IBM WebSphere Application Server Liberty Core|IBM WebSphere Hybrid Edition|
|        |X       |       |X                 |IBM WebSphere Application Server Network Deployment|IBM Cloud Pak for Applications|
|        |X       |       |X                 |IBM WebSphere Application Server Network Deployment|IBM WebSphere Application Server Family Edition|

Feel free to do a live testing with [preview link](https://portal.azure.com/?feature.customportal=false#create/microsoft_javaeeonazure_test.20191212-arm-rhel-was-nd-v9-cluster-previewazure-liberty-aks).

## UI changes review

Besides visiting [preview link](https://portal.azure.com/?feature.customportal=false#create/microsoft_javaeeonazure_test.20191212-arm-rhel-was-nd-v9-cluster-previewazure-liberty-aks) for reviewing the renamed blade `Operator and application` (which comes from `Configure application`), reviewer can preview the UI changes using the **Create UI Definition Sandbox**:

- Open [Create UI Definition Sandbox](https://portal.azure.com/?feature.customPortal=false#blade/Microsoft_Azure_CreateUIDef/SandboxBlade) in a new tab of browser.
- Open UI source code [createUiDefinition.json](https://raw.githubusercontent.com/majguo/azure.liberty.aks/support-wlo/src/main/arm/createUiDefinition.json) in a new tab of the browser.
- Replace the content of work area in **Create UI Definition Sandbox** with the content of **UI source code `createUiDefinition.json`**.
- Click **Preview** at the left-bottom of **Create UI Definition Sandbox**.
- The UI changes in the renamed `Operator and application` blade include:
   - Add a new option **IBM supported? Yes/No**.
   - If selecting **Yes**, a checkbox **Accept license** and two more options **Edition** and **Product entitlement source** are displayed for user inputs. 

Also attached the screenshots for quick overview:

- **IBM supported: No (default)**
  image

- **IBM supported: Yes**
  image

Signed-off-by: Jianguo Ma <jiangma@microsoft.com>

### CODE REVIEW

File: .github/workflows/integration-test.yaml [hunk 1]
```diff
@@ -7,35 +7,33 @@ on:
...
+        required: true
+        type: boolean
+        default: true
       deleteAzureResources:
         description: 'Delete Azure resources at the end'
         required: true
         type: boolean
         default: true
-  # Allows you to run this workflow using GitHub APIs
+  # Allows you to run this workflow using GitHub workflow dispatch APIs
```
Comment (edburns) [inline .github/workflows/integration-test.yaml:25]: Good clarity.

**Review by edburns — APPROVED** [review]

**Comment by git4rk** [comment]
@majguo I have a couple of questions:
1. Are we using OLM to install the WLO?
2. If customer choses to use `IBM Supported`, will we still provide sample application CR yaml for WebSphereLibertyApplication?

**Comment by majguo** [comment]
@git4rk 
1. We're using [kustomize](https://www.ibm.com/docs/en/was-liberty/base?topic=cli-installing-kustomize) to both WLO/OLO.
1. Yes. See [src/main/scripts/websphere-liberty-application-agic.yaml.template](https://github.com/WASdev/azure.liberty.aks/pull/69/files#diff-8161414ff31e5269ab759414e243fc42326f9f8ae6c1b335404bc6c5841c51fe) and [websphere-liberty-application-agic.yaml.template](https://github.com/WASdev/azure.liberty.aks/pull/69/files#diff-8161414ff31e5269ab759414e243fc42326f9f8ae6c1b335404bc6c5841c51fe).

File: .github/workflows/integration-test.yaml [hunk 2]
```diff
@@ -7,35 +7,33 @@ on:
         required: true
         type: boolean
         default: true
+      deployWLO:
+        description: 'WebSphere Liberty operator'
```
Comment (git4rk) [inline .github/workflows/integration-test.yaml:11]: WebSphere Liberty operator -> WebSphere Liberty Operator

Comment (edburns) [reply .github/workflows/integration-test.yaml]: Fixed.

Comment (edburns) [reply .github/workflows/integration-test.yaml]: @git4rk

**Review by edburns — APPROVED** [review]

**Comment by git4rk** [comment]
@edburns Can you please review the recent changes. Thanks.

**Review by edburns — APPROVED** [review]

**Review by git4rk — APPROVED** [review]

**Pull request merged**